app/models/ems_event.rb

```diff
@@ -90,7 +90,17 @@ def self.process_vm_in_event!(event, options = {})
   end
 
   def self.process_host_in_event!(event, options = {})
+    uid_ems = event.delete(:host_uid_ems)
     process_object_in_event!(Host, event, options)
+
+    if event[:host_id].nil? && uid_ems.present?
+      # Attempt to find a host in the current EMS first, then fallback to archived hosts
+      host = Host.find_by(:ems_id => event[:ems_id], :uid_ems => uid_ems) || Host.find_by(:ems_id => nil, :uid_ems => uid_ems)
```

Comment:
Wasn't sure if we should be looking at archived hosts also but the `process_vm_in_event!` method explicitly does look for vms that aren't in the current EMS.

That method does a simple `VmOrTemplate.find_by(:uid_ems => uid_ems)` which seems dangerous considering we know that uid_ems's aren't unique so just grabbing the first might not be accurate.

I thought it was better to try to find a host in the current EMS first, then fallback to explicitly archived hosts (aka not possibly belonging to a different provider).

Reply:
You can single query this with:

```ruby
Host.where(:uid_ems => uid_ems, :ems_id => [event[:ems_id], nil]).order(:ems_id).first
```

Since nulls get sorted last, this works

Reply:
if we want to be even more explicit about the nulls last, we can do:

```ruby
Host.where(:uid_ems => uid_ems, :ems_id => [event[:ems_id], nil]).order("ems_id NULLS LAST").first
```

Reply:
:+1: done and added spec for the case of duplicate uid_ems one archived one active